smithy-build/src/main/java/software/amazon/smithy/build/transforms/FilterSuppressions.java

```diff
@@ -293,25 +295,24 @@ private void filterSuppressionTraits(
             }
 
             if (allowed.isEmpty()) {
-                builder.addShape(Shape.shapeToBuilder(shape).removeTrait(SuppressTrait.ID).build());
+                replacementShapes.add(Shape.shapeToBuilder(shape).removeTrait(SuppressTrait.ID).build());
             } else if (!allowed.equals(trait.getValues())) {
                 trait = trait.toBuilder().values(allowed).build();
-                builder.addShape(Shape.shapeToBuilder(shape).addTrait(trait).build());
+                replacementShapes.add(Shape.shapeToBuilder(shape).addTrait(trait).build());
             }
         }
+        return transformer.replaceShapes(model, replacementShapes);
     }
 
-    private void filterMetadata(
+    private Model filterMetadata(
```

Comment:
minor: Can you add a check to make sure the model has suppressions defined before recreating it?

Reply:
Good suggestion. Added a check to make sure suppressions changed before recreating. Also added couple tests cases for 1. no suppressions 2. no change to suppressions.